src/track/beats.h

```diff
@@ -166,10 +173,21 @@ class Beats {
     virtual BeatsPointer scale(BpmScale scale) const = 0;
 
     /// Adjust the beats so the global average BPM matches `bpm`.
-    virtual BeatsPointer setBpm(mixxx::Bpm bpm) = 0;
+    virtual BeatsPointer setBpm(mixxx::Bpm bpm) const = 0;
 
   protected:
+    /// Type tag for making public constructors of derived classes inaccessible.
+    ///
+    /// The constructors must be public for using std::make_shared().
+    struct MakeSharedTag {};
```

Comment:
It is only needed for make_shared(). Otherwise the constructors could be hidden by making them private.

Reply:
I disagree. Redundantly repeating the type name of the type tag class where it is not needed is useless. It must only specified explicitly when using std::make_shared(), the only purpose it was introduced for.

Reply:
We hit often such discussion in our reviews. I think this is sourced by our different roles.
For me as a reviewer it is important to understand the code at once without looking into other references. 
You as the author are more interested to avoid redundant statements. 
Please keep in mind that the human reader is not a compiler which reads thousands of lines is a second, fetched form many included files. To make the code readable I prefer redundant statements more than explaining comments, because the compiler can check them. 

A plain {} as a parameter can be everything, the reader needs to look up other references to understand it. This is the same for bool function parameter by the way. 

Whenever the function signature changes, a {} will still compile even though it might be wrong.

Reply:
The plan was to use `{}` everywhere but `std::make_shared()` fails to deduce the type.

I don't see a reason to re-introduce the redundant type name in private forwards.